### DISCUSSION TITLE
Each Github Pages site should have its own subdomain for security (and other modern web dev) reasons

### DISCUSSION THREAD

**Discussion by josephrocca** [community/community#60479] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

#### Problem
I like using Github Pages for little tools and apps, but some of these apps require storing user data in `localStorage` and `IndexedDB`. For example, https://josephrocca.github.io/OpenCharacters/ has chat logs between the LLM and the user stored. The more tools/apps I create, the more dangerous a bug becomes, since it could mean that sensitive user data can be exfiltrated by simply sharing a link with one another. For example, I often store app parameters in the URL hash - for example as a way to share characters on OpenCharacters. Someone may figure out a bug that allows them to share a URL with someone which triggers a request to an external server that carries the user's data.

Also, modern web app security & features expect that each app/site has its own origin. For example PWA manifests are specific to an origin, so it's impossible for me to make a Github Pages site (like OpenCharacters) into a PWA.

#### Solution
Each Github Pages site should have its own subdomain. This seems like a complete no-brainer, and the current state of affairs is basically a bug / security issue. This isn't a "nice to have" feature for Github Pages - it's essential.

 * Old: https://josephrocca.github.io/OpenCharacters
 * New: https://opencharacters.josephrocca.github.io

Note that you'll need to keep the existing ability to create an app under the user subdomain, since **this will be needed for backwards compatibility** on many sites. You'd just give people the option to switch over to the new approach, and make it use that by default. I'd imagine that you'll also need to allow people to specify a custom subdomain name since maybe it's possible to have a repo name that's not a valid subdomain (though I'm just guessing).

Related: https://github.com/orgs/community/discussions/40683

#### Non-Solutions

* **Buy a domain and link it with the custom domains feature**: The reason I use Github Pages is because I want my site/app to "last forever" - not until I get tired of paying for my app's domain.
* **Create a separate org for each app/site**: I create lots of little experiments, and having an org for each one would be cumbersome. The ability to turn a repo into a static site with a couple of clicks is the main appeal of Github Pages.

#### Is anyone in charge of Pages?
I'm kind of guessing that this issue will go unnoticed. Radio silence on the most upvoted (as of writing) issue of all time in the [Pages category](https://github.com/orgs/community/discussions/categories/pages?discussions_q=is%3Aopen+category%3APages+sort%3Atop) (which I posted over a year ago):

 * "Allow setting COOP and COEP headers in Github Pages" - https://github.com/orgs/community/discussions/13309

@queenofcorgis, I don't know who to tag here - can you please tag the relevant people? Is anyone in charge of Pages right now?

**Comment by yoannchaudet** [comment]
Hello,

Pages team here.

We have considered this feature. Unfortunately we have no plans to implement it in the short term.
